src/Core/Content/Category/CategoryException.php

```diff
@@ -27,6 +27,11 @@ public static function categoryNotFound(string $id): ShopwareHttpException
         return new CategoryNotFoundException($id);
     }
 
+    public static function categoryIdIsNotValidHex(string $categoryId): CategoryIdIsNotValidHexException
+    {
+        return new CategoryIdIsNotValidHexException($categoryId);
```

Comment:
as we do not catch `CategoryIdIsNotValidHexException` ourselves, please only return an instance of `CategoryException`. have a look a the ADR for more information about what to implement when: https://github.com/shopware/shopware/blob/trunk/adr/2022-02-24-domain-exceptions.md